### PR TITLE
Trino file system cleanups

### PR DESCRIPTION
**Pull request by electrum**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCreateTableFailure.java [hunk 1]
```diff
@@ -170,7 +170,7 @@ protected void assertMetadataLocation(String tableName, boolean shouldMetadataFi
         boolean metadataFileFound = false;
         while (fileIterator.hasNext()) {
             FileEntry fileEntry = fileIterator.next();
-            String path = fileEntry.path();
+            String path = fileEntry.location();
```
Comment (dain) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCreateTableFailure.java:173]: location

File: lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java [hunk 2]
```diff
@@ -75,11 +76,11 @@ public FileEntry next()
...
         }
 
-        List<BlockLocation> locations = Stream.of(status.getBlockLocations())
-                .map(HdfsFileIterator::toTrinoBlockLocation)
+        List<Block> locations = Stream.of(status.getBlockLocations())
+                .map(HdfsFileIterator::toTrinoBlock)
                 .collect(toImmutableList());
 
-        Optional<List<BlockLocation>> blockLocations = locations.isEmpty() ? Optional.empty() : Optional.of(locations);
+        Optional<List<Block>> blockLocations = locations.isEmpty() ? Optional.empty() : Optional.of(locations);
```
Comment (dain) [inline lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java:83]: still `blockLocations`

File: lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java [hunk 3]
```diff
@@ -75,11 +76,11 @@ public FileEntry next()
             path += relativeUri.getPath();
         }
 
-        List<BlockLocation> locations = Stream.of(status.getBlockLocations())
-                .map(HdfsFileIterator::toTrinoBlockLocation)
+        List<Block> locations = Stream.of(status.getBlockLocations())
```
Comment (dain) [inline lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java:79]: rename get locations?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/BlockLocation.java [hunk 4]
```diff
@@ -45,11 +45,11 @@ public static List<BlockLocation> fromHiveBlockLocations(@Nullable org.apache.ha
                 .collect(toImmutableList());
     }
 
-    public BlockLocation(FileEntry.BlockLocation blockLocation)
+    public BlockLocation(FileEntry.Block block)
```
Comment (dain) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/BlockLocation.java:48]: Should this just be block?

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java [hunk 5]
```diff
@@ -215,7 +215,7 @@ private void doVacuum(
         FileIterator listing = fileSystem.listFiles(tableLocation.toString());
         while (listing.hasNext()) {
             FileEntry entry = listing.next();
-            String path = entry.path();
+            String path = entry.location();
```
Comment (dain) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java:218]: location

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java [hunk 6]
```diff
@@ -214,7 +214,7 @@ private List<String> listAllFilesRecursive(String directory)
             FileIterator files = fileSystem.listFiles(path);
             while (files.hasNext()) {
                 FileEntry file = files.next();
-                paths.add(file.path());
+                paths.add(file.location());
```
Comment (dain) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java:217]: locations?

File: lib/trino-hive-formats/src/main/java/io/trino/hive/formats/DataSeekableInputStream.java [hunk 7]
```diff
@@ -58,12 +58,12 @@
 
     private int bufferFill;
 
-    public DataSeekableInputStream(SeekableInputStream inputStream)
+    public DataSeekableInputStream(TrinoInputStream inputStream)
```
Comment (dain) [inline lib/trino-hive-formats/src/main/java/io/trino/hive/formats/DataSeekableInputStream.java:61]: No seekable

File: lib/trino-hive-formats/src/test/java/io/trino/hive/formats/TestDataTrinoInputStream.java [hunk 8]
```diff
@@ -727,7 +727,7 @@ protected abstract String readActual(DataSeekableInputStream input)
 
     private static DataSeekableInputStream createDataSeekableInputStream(byte[] bytes)
```
Comment (dain) [inline lib/trino-hive-formats/src/test/java/io/trino/hive/formats/TestDataTrinoInputStream.java:728]: No seekable

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/fileio/ForwardingSeekableInputStream.java [hunk 9]
```diff
@@ -23,9 +24,9 @@
 public class ForwardingSeekableInputStream
         extends SeekableInputStream
 {
-    private final io.trino.filesystem.SeekableInputStream stream;
+    private final TrinoInputStream stream;
 
-    public ForwardingSeekableInputStream(io.trino.filesystem.SeekableInputStream stream)
+    public ForwardingSeekableInputStream(TrinoInputStream stream)
```
Comment (dain) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/fileio/ForwardingSeekableInputStream.java:29]: No seekable

**Review by dain — APPROVED** [review]
you missed a few variable names

File: lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java (hunk 3 above)
Comment (electrum) [reply lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java:79]: That's in `org.apache.hadoop.fs.LocatedFileStatus`

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/fileio/ForwardingSeekableInputStream.java (hunk 9 above)
Comment (electrum) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/fileio/ForwardingSeekableInputStream.java:29]: This is forwarding to Iceberg

File: lib/trino-hive-formats/src/main/java/io/trino/hive/formats/DataSeekableInputStream.java (hunk 7 above)
Comment (electrum) [reply lib/trino-hive-formats/src/main/java/io/trino/hive/formats/DataSeekableInputStream.java:61]: I added another commit to rename `DataSeekableInputStream` to `TrinoDataInputStream`

**Review by dain — APPROVED** [review]

**Pull request merged**